src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,207 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace creators {
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief A Domain for testing purposes.
```

Comment:
In previous discussions with @nilsdeppe and perhaps others we talked about having the BinaryCompactObject domain be the most configurable binary domain in terms of the options, and that we will have specialized domains for BBH/BNS/BHNS so that we don't expose unnecessary options that might be distracting to users who are less familiar with configuring domains. This way as well we can give more appropriate option names and more specific descriptions in the more specialized domains.

Reply:
I would still change 'A Domain for testing purposes' to 'A general Domain for two compact objects'.